src/odemis/driver/pwrmccdaq.py

```diff
@@ -0,0 +1,420 @@
+import logging
+import time
+from threading import Thread
+from typing import List, Tuple, Dict, Iterable
+
+import odemis
+from odemis import model
+from odemis.driver._mccdaq import usb_1208LS
+from odemis.model import HwError, Emitter, HwComponent
+
+MAX_VOLTAGE_VALUE = 0x3ff  # for 1208LS
+MAX_VOLTAGE = 5.0  # V
+INTERLOCK_POLL_INTERVAL = 0.1
+
+
+class MCCDevice(HwComponent):
+    """
+    Basic version of the Class for MCC DAQ device functionality with support for digital input status.
+    This class can be inherited for more functionality to support more complex components.
+    """
+
+    def __init__(self, name: str, role: str, mcc_device: str,
+                 di_channels: Dict[int, Tuple[str, bool]] = None, **kwargs):
+        """
+        :param mcc_device (str or None): serial number or None (null in yaml file) for auto-detect.
+            the device is considered a USB-powered MCC DAQ 1208LS device.
+        :param di_channels (dict -> int, list(str, bool)):
+            the DIO channel used for TTL signal status change through the MCC device.
+            for example di_channels: {2: ["interlockTriggered", False], 7: ["leftSwitch", True]}.
+            key is the channel number(int), value is a list with VA name(str) and tll high is true(bool),
+                this means that the VA will be considered True or False when the TTL signal is high.
+            with this approach it is actually possible to keep track of customized status changes.
+        """
+        super().__init__(name, role, **kwargs)
+        # force add a trailing 0 or just add a comment to config file?
+        # if device and len(device) < 8:
+        #     device = "0" + device
+        self._name = name
+        self._di_channels = di_channels if di_channels else {}
+        self.device = None
+        self._status_thread = None
+        self._channel_vas = {}
+
+        if mcc_device == "fake":
+            self.device = MCCDeviceSimulator()
+        else:
+            try:
+                self.device = usb_1208LS(mcc_device)
+            except HwError:
+                raise HwError("Failed to open MCC DAQ device with s/n '%s'" % mcc_device)
+
+        for channel, va_properties in self._di_channels.items():
+            try:
+                # check if va_properties is an iterator with 2 elements
+                va_name, ttl_high = va_properties
+            except ValueError:
+                raise ValueError("di_channels expects for each channel a "
+                                 "[int, [str, [boolean]]], but got %s" % (va_properties,))
+
+            # check the di_channel values
+            try:
+                self.channel_to_port(channel)
+            except ValueError:
+                raise ValueError(f"Invalid channel value for status {self._di_channels[channel]}")
+
+            # create a VA with False as default
+            va = model.BooleanVA(False, readonly=True)
```

Comment:
```suggestion
            va = model.BooleanVA(ttl_high, readonly=True)
```
Maybe I am wrong

Reply:
I guess its quite confusing but we like the VA default value to start with False. The ttl_high boolean represents the interpretation of the BooleanVA as in which case to set it to true. When ttl_high is true the BooleanVA should still be false when the signal bit is 0. When the bit changes to 1, the ttl signal is high and the BooleanVA set to true. This way the bit can also be 1 and the BooleanVA be false but then ttl_high should be false of course.